The `WKInteractionRegionDebugFill` user default should be cached like the others

etiennesegonzac opened the pull request:
#### 9e1971d5e04a29156280f2f3921ff0776f59a7cc
<pre>
The `WKInteractionRegionDebugFill` user default should be cached like the others
<a href="https://bugs.webkit.org/show_bug.cgi?id=292281">https://bugs.webkit.org/show_bug.cgi?id=292281</a>
&lt;<a href="https://rdar.apple.com/150200937">rdar://150200937</a>&gt;

Reviewed by Tim Horton and Simon Fraser.

It&apos;s cheap but in a hot path. Use the same static/dipatch_once method
we use for `WKInteractionRegionBrightnessMultiplier` to expose
`WKInteractionRegionDebugFill`.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionEffectUserInfo):
(WebKit::brightnessMultiplier):
Remove needless dipatch_once.
(WebKit::applyBackgroundColorForDebugging):
(WebKit::updateLayersForInteractionRegions):

Canonical link: <a href="https://commits.webkit.org/294426@main">https://commits.webkit.org/294426@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/176d9aa2da916013e0ded5320a1dc18da19a79e9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101778 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21446 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11762 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106936 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52412 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21754 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/29969 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77493 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34515 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104785 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16807 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91907 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57831 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16632 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51763 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86488 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10002 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109293 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28911 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21291 "Exiting early after 60 failures. 53833 tests run. 60 failures") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86472 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29272 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88108 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86043 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21896 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30797 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8518 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23080 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28839 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34129 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28650 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31973 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30209 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
```diff
@@ -76,6 +76,17 @@ static float brightnessMultiplier()
+{
+    static bool applyBackground = false;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
```
dispatch once seems needless, isn't this always called on the main thread?

etiennesegonzac committed:
The `WKInteractionRegionDebugFill` user default should be cached like the others
https://bugs.webkit.org/show_bug.cgi?id=292281
<rdar://150200937>

Reviewed by Tim Horton and Simon Fraser.

It's cheap but in a hot path. Use the same static/dipatch_once method
we use for `WKInteractionRegionBrightnessMultiplier` to expose
`WKInteractionRegionDebugFill`.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionEffectUserInfo):
(WebKit::brightnessMultiplier):
Remove needless dipatch_once.
(WebKit::applyBackgroundColorForDebugging):
(WebKit::updateLayersForInteractionRegions):

Canonical link: https://commits.webkit.org/294426@main
